appointment/src/main/java/com/bmwapp/appointment/service/AppointmentService.java

```diff
@@ -63,15 +71,37 @@ public Appointment updateAppointment(Appointment appointment) {
         Provider<Appointment> appointmentProvider = p -> oldAppointment;
         propertyMapper.setProvider(appointmentProvider);
 
+        if(contextProvider.userHasRole("CUSTOMER")){
+            modelMapper.getTypeMap(Appointment.class, Appointment.class)
+                    .addMappings(mapper ->mapper.skip(Appointment::setCustomerId))
+                    .addMappings(mapper ->mapper.skip(Appointment::setFlatId))
+                    .addMappings(mapper ->mapper.skip(Appointment::setAppointmentName));
+        } else {
+            modelMapper.getTypeMap(Appointment.class, Appointment.class)
+                    .addMappings(mapper ->mapper.map(Appointment::getCustomerId, Appointment::setCustomerId))
+                    .addMappings(mapper ->mapper.map(Appointment::getFlatId, Appointment::setFlatId))
+                    .addMappings(mapper ->mapper.map(Appointment::getAppointmentName, Appointment::setAppointmentName));
+        }
+
         Appointment updatedAppointment = modelMapper.map(appointment, Appointment.class);
         return appointmentRepository.save(updatedAppointment);
     }
 
-    public void deleteAppointmentByIdAndCustomerId(Integer appointmentId, String customerId) {
-        Appointment appointment = appointmentRepository
-                .findById(appointmentId)
-                .orElseThrow(() -> new ResourceNotFoundException(String.format("Appointment with id %d not found", appointmentId)));
-        appointmentRepository.deleteByIdAndCustomerId(appointmentId, customerId);
+    public void deleteAppointment(Integer appointmentId) {
+        if (!appointmentRepository.findById(appointmentId).isPresent() || (contextProvider.userHasRole("CUSTOMER")
```

Comment:
do we need more than "appointmentRepository.findById(appointmentId).isPresent()" in this input? If it's not present we could throw that it's not found.

We check the user role below

Reply:
Potrzebujemy, sprawdzam najpierw czy istnieje taki appointment niezaleznie od roli. Jesli istnieje i rola to agent, to jest ok. Jesli istnieje i jest to customer to jeszcze musze sprawdzic czy istnieje appointment jeszcze z ID Customera.
Jesli tak to ok. Jesli nie to rzuca blad.